File: frontend/src/lib/texto.ts
/** Normalização de texto para busca/comparação no cliente.
 *
 * Espelha `app/services/texto.py::normalizar_texto` — minúsculo, sem acento, espaços colapsados.
 * Em pt-BR comparar com acento erra o caso mais comum ("Farmácia" vs. "farmacia").
 */
export function normalizarBusca(valor: string | null | undefined): string {
  return (valor ?? "")
    .normalize("NFKD")
    .replace(/\p{Diacritic}/gu, "")
    .toLowerCase()
    .trim()
    .split(/\s+/)
    .join(" ")
}

File: app/services/texto.py
"""Normalização de texto para comparação (§4.5).

Usada em dois lugares que precisam casar exatamente do mesmo jeito: a unicidade do nome de uma
categoria personalizada e o casamento das regras de categorização. Como o vocabulário é pt-BR,
comparar sem remover acento erraria o caso mais comum ("Farmácia" vs. "farmacia").

NÃO confundir com `assinatura_deteccao.normalizar_nome`, que só faz minúsculo + colapso de espaço:
aquela define o casamento de aliases de assinatura e mudá-la alteraria vínculos já existentes.
"""

import unicodedata


def normalizar_texto(valor: str | None) -> str:
    """Minúsculo, sem acento, espaços colapsados. `None`/vazio → `""`."""
    sem_acento = unicodedata.normalize("NFKD", (valor or "").strip())
    sem_acento = "".join(c for c in sem_acento if not unicodedata.combining(c))
    return " ".join(sem_acento.lower().split())
